pkg/gatewayserver/io/basicstationlns/basicstationlns.go

```diff
@@ -75,24 +105,25 @@ func (s *srv) handleDiscover(c echo.Context) error {
 	ws, err := s.upgrader.Upgrade(c.Response(), c.Request(), nil)
 	if err != nil {
 		logger.WithError(err).Debug("Failed to upgrade request to websocket connection")
-		return err
+		return nil
```

Comment:
Why?

Reply:
At this point, the websocket is not upgraded, so we should be able to return a normal error response here.

About those echo errors; let's handle that in an error handler and still return errors here.